### PR TITLE
Allow paths as filenames in multipart/form-data requests

### PR DESCRIPTION
**Pull request by mneil**
I was trying to use this in conjunction with loopback container service which accepts a path as a filename when connecting to S3.

Long story short, the plugin coerces the filename to the files basename — so I can't send a file path as the name like /path/to/file.png. 

I tried reading through as much of https://www.ietf.org/rfc/rfc2388.txt https://tools.ietf.org/html/rfc2184 as I could and as far as I can tell the only requirements for the filename are that it is US-ASCII. I like how easy your plugin makes it to send file data as multipart/form-data but I would argue it's the application developers job to determine the logic behind properties such as the filename.

I'd hate to break this for others so maybe adding an option to use basename for now would be good?

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/form-data/form-data/pull/181" target="_blank">ReviewNinja

File: lib/form_data.js [hunk 1]
```diff
@@ -45,6 +45,8 @@ FormData.prototype.append = function(field, value, options) {
   if (typeof options == 'string') {
     options = {filename: options};
   }
+  if( !options.hasOwnProperty('basename') )
+    options.basename = true;
```
Comment (alexindigo) [inline lib/form_data.js]: Codacy Issue found: [Expected '{' and instead saw 'options'.](https://www.codacy.com/app/form-data/form-data/file/2240808266/issues/source?bid=2805153&fileBranchId=3179066#l49)

**Comment by mneil** [comment]
I should have left this well enough alone. I squashed my commits down but ended up rewriting some of your history where I didn't mean to. I can close this and open another with the correct commit if you'd like. Let me know your preference for PRs

**Comment by sebdeckers** [comment]
Closing because this PR as the feature has been implemented in #203 which shipped as `v2.2.0`. Use the `filepath` option to pass subdirectory information.

**Pull request closed without merging**